test/qa/lib/daemon.py

```diff
@@ -71,14 +76,18 @@ def uninstall_peer(ssh_client: ssh.Ssh):
 
 def start():
     """Starts daemon and blocks until it is actually started."""
-    if is_init_systemd():
-        sh.sudo.systemctl.start.nordvpnd()
-        return
-    # call to init.d returns before the daemon is actually started
-    _rewrite_log_path()
-    sh.sudo("/etc/init.d/nordvpn", "start")
-    while not is_running():
-        time.sleep(1)
+    if is_under_snap():
+        #sh.sudo.snap("start", "nordvpn")
+        os.popen("sudo snap start nordvpn").read()
+    elif is_init_systemd():
+        #sh.sudo.systemctl.start.nordvpnd()
+        os.popen("sudo systemctl start nordvpn").read()
+    else:
+        # call to init.d returns before the daemon is actually started
+        _rewrite_log_path()
+        sh.sudo("/etc/init.d/nordvpn", "start")
+        while not is_running():
```

Comment:
shouldn't while check for all types that the daemon started?

Reply:
Before my changes, `waiting` for daemon start was only for `init.d` - maybe there was a reason for that, so, I left the same way.

Reply:
Probably because it never ran on systemd env :smile: